vault/core.go

```diff
@@ -1001,21 +1001,8 @@ func CreateCore(conf *CoreConfig) (*Core, error) {
 		effectiveSDKVersion = version.GetVersion().Version
 	}
 
-	var detectDeadlocks []string
-	if conf.DetectDeadlocks != "" {
-		detectDeadlocks = strings.Split(conf.DetectDeadlocks, ",")
-		for k, v := range detectDeadlocks {
-			detectDeadlocks[k] = strings.ToLower(strings.TrimSpace(v))
-		}
-	}
-
-	// Use imported logging deadlock if requested
-	var stateLock locking.RWMutex
-	stateLock = &locking.SyncRWMutex{}
-
-	if slices.Contains(detectDeadlocks, "statelock") {
-		stateLock = &locking.DeadlockRWMutex{}
-	}
+	detectDeadlocks := parseDetectDeadlockConfigParameter(conf.DetectDeadlocks)
+	stateLock := createAppropriateRWMutex(detectDeadlocks, "statelock")
```

Comment:
Minor nit/bikeshed -- feel free to ignore:

I'm not sure `Appropriate` helps a lot as a descriptor in this method. How do you feel about dropping that in favor of `createRWMutex`. You already have a really nice godoc for the method, which should clarify what's going on here.

Reply:
Totally agree! I was struggling on a good name and am open to suggestions.

Reply:
My latest commit is suggesting `CreateConfigurableRWMutex` as a function name.